[WIP] Atomic Swaps

ChrisCates opened the pull request:
### Description

Currently, a work in progress pull request to just keep track of progress on Atomic Swaps.
Right now, all that is in the pull request is a regression testing environment for both Divi and Bitcoin.

If you want to check it out, you can run:

```bash
git remote add chriscates https://github.com/ChrisCates/Divi
git fetch chriscates
git checkout chriscates/@ChrisCates/AtomicSwaps
```

### Requirements

Review issue #122 for requirements.

### Features

**Current features currently in PR ready to review.**

- A regression testing environment for Bitcoin and Divi.

99darwin commented:
This looks fine, although, because we will likely have several of these for various features, I think we should add these tests to a specific folder. Whether that's the preexisting `test`, `docker`, or some more specific directory.

ChrisCates commented:
Absolutely!
Perhaps `test/atomic-swaps/test.Dockerfile` would make sense.

galpHub commented on atomic.test.Dockerfile:
```diff
@@ -0,0 +1,34 @@
+FROM ubuntu:bionic
+
+RUN apt-get update
```
```suggestion
RUN apt-get update
RUN apt-get install git -y
```

galpHub commented on atomic.test.Dockerfile:
```diff
@@ -0,0 +1,34 @@
+RUN make
+
+RUN git clone https://github.com/bitcoin/bitcoin
+RUN cd bitcoin
```
replace 29-32 by:
RUN cd bitcoin && ./autogen.sh && ./configure && make
or just line 29 by:
WORKDIR /app/bitcoin

ChrisCates commented:
@galpHub, I've added documentation in regards to setting up the regression testing environment for Atomic Swaps.

https://github.com/DiviProject/Divi/blob/2c6cee57aaaf40aea7dab5f960e3d5a26abcc516/test/docker/ATOMIC.ENV.MD

Will be working on some test cases for Atomic Swaps now.

ChrisCates commented:
Just added instructions to configure Atomic Swap code up.
Also added batch and shell scripts to configure for build with golang.

https://github.com/DiviProject/Divi/blob/12db5b517df8961a781f253fcb97ac57ad5c9caa/src/atomic-swap/README.md

ChrisCates commented:
All work will be migrated to a separate repo: https://github.com/DiviProject/atomic-swaps
We need to do some large scale refactoring to gRPC in order to get this shippable to production.

Closing this PR.
